Fix fragile payment gateway names

lovo-h opened the pull request:
<!--
Did I add a title? A descriptive, yet concise, title.
-->

<!--
Issue: Link to the GitHub issue this PR addresses (if appropriate).
-->

## Changes proposed in this Pull Request:

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for the PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

This pull request (PR) removes some fragility in the code where array keys were used to display payment gateway names. Instead, a label property was added to the `WC_Stripe_Payment_Gateway` class, in order to store each respective payment gateway's name.

Note that tests do not exist for this minor change.

## Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Please follow the following guidelines when writing testing instructions:

- Include screenshots if there is no similar flow in the critical flows: https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Critical-flows
- Assume instructions will be copied over to the Release Testing Instructions wiki page: https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions
- Assume instructions will be followed by external testers.
- Assume tester does not have intimate knowledge of Stripe.
-->

#### Make sure payment gateway names are displayed properly capitalized

- As a merchant, go to `WooCommerce > Settings > General`
- Make sure that `Currency`, under `Currency options`, displays "United States (US) dollar ($)"
- Click `Save changes` if necessary
- Go to `WooCommerce > Settings > Payments`
- Click on the `Stripe` method and then go to the `Settings` tab
- Expand `Advanced settings` (at the bottom)
- Make sure that `Try the new checkout experience (early access)` is disabled
- Click `Save changes` if necessary
- Go to `WooCommerce > Settings > Payments` and enable the following methods
  - `Stripe`, `Stripe SEPA Direct Debit`, `Stripe giropay`, `Stripe iDEAL`, `Stripe Bancontact`, `Stripe EPS`, `Stripe Sofort`, and `Stripe P24`
- Refresh the page and make sure that the payment gateway names are displayed properly capitalized:
  - `Bancontact`, `EPS`, `giropay`, `iDEAL`, `P24`, `SEPA`, and `Sofort` (see image below)

![image](https://user-images.githubusercontent.com/6844516/145728410-08ff6f35-d88f-4521-a1e8-710462b77e0d.png)

#### Make sure payment method names are displayed properly capitalized

- As a merchant, go to `WooCommerce > Settings > General`
- Make sure that `Currency`, under `Currency options`, displays "United States (US) dollar ($)"
- Click `Save changes` if necessary
- Go to `WooCommerce > Settings > Payments`
- Click on the `Stripe` method and then go to the `Settings` tab
- Expand `Advanced settings` (at the bottom)
- Make sure that `Try the new checkout experience (early access)` is enabled
- Click `Save changes` if necessary
- Go to `WooCommerce > Settings > Payments`
- Click on the `Stripe` method and then go to the `Payment Methods` tab
- Enable the following payment methods
  - `giropay`, `EPS`, `Bancontact`, `iDEAL`, `SEPA Direct debit payment`, `PRzelewy24`, and `Sofort`
- Click `Save changes` if necessary
- Refresh the page and make sure that the payment gateway names are displayed properly capitalized:
  - `giropay`, `EPS`, `Bancontact`, `iDEAL`, `SEPA`, `Przelewy24`, and `Sofort` (see image below)

![image](https://user-images.githubusercontent.com/6844516/145728537-ca209395-9137-4392-80ba-2f2ea74bdcfe.png)

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry (or does not apply)
-   [x] Tested on mobile (or does not apply)

**Post merge**

-   [x] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)

lovo-h committed:
Use payment gateway label instead of array key, for gateway name

Before, payment gateway names were retrieved from array keys. This made
the code a bit fragile and required lowercasing the name for some use-
cases. Now, a label property was added into WC_Stripe_Payment_Gateway
class and that property is used to store the payment gateway
name, instead.

lovo-h committed:
Remove redundant lowercase call for UPE payment method names

Before, payment method names were being lowercased, even though the name
was already lowercase in all the respective payment method classes.

rafaelzaleski commented on includes/abstracts/abstract-wc-stripe-payment-gateway.php:
```diff
@@ -24,6 +24,22 @@ abstract class WC_Stripe_Payment_Gateway extends WC_Payment_Gateway_CC {
+	 *
+	 * @var string
+	 */
+	protected $label;
```
Since the label property will only be used to show admin notices, we can remove it and use the `method_title` property instead. It wouldn't make a difference for the store admin to see "Stripe Oxxo" instead of "Oxxo" in the message.

lovo-h replied:
I started off by using the `get_method_title()` method but I noticed it read significantly different than when UPE was enabled. So, I opted for adding the `get_label()` method, in order to maintain some naming consistency.

I just want to highlight that by using the `get_method_title()`, we get the following:

![image](https://user-images.githubusercontent.com/6844516/145836272-2f2f4a65-df6c-4bd8-a504-6bff23966bf7.png)

For comparison, when we enable UPE, we get the following:

![image](https://user-images.githubusercontent.com/6844516/145728537-ca209395-9137-4392-80ba-2f2ea74bdcfe.png)

I agree that it's easier to maintain the code if we use the `method_title` property so I will go ahead and make the change 👍

lovo-h committed:
Reuse method_title property, as opposed to adding a label property

Because the WC_Stripe_Payment_Gateway class already had a `method_title`
property that has the name of the payment gateway, it was a bit
redundant to add the `label` property. This commit removes the `label`
property and reuses the `method_title` property, instead. This does mean
that the payment method/gateway will read slightly different when UPE is
enabled/disabled.

See the following GH comment for more info:
https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2230#discussion_r767743479

rafaelzaleski reviewed (approved):
Looks good. Thanks! :shipit:
